compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertCores.cpp

```diff
@@ -111,6 +112,16 @@ LogicalResult insertCoreOps(mlir::ModuleOp moduleOp) {
       } else if (auto linalgOp = dyn_cast<linalg::LinalgOp>(op)) {
         rewriter.setInsertionPoint(endOp);
         rewriter.moveOpBefore(linalgOp, endOp);
+      } else if (isa<vector::ContractionOp>(op)) {
+        // Because vector.contract op is surrounded by vectorized loop nest, we
+        // need to traverse to the outermost loop to move the entire vectorized
+        // computation within amdaie.core op.
+        Operation *currOp = op;
+        while (currOp->getParentOp() != forallOp) {
+          currOp = currOp->getParentOp();
```

Comment:
I think you could just use currOp = forallOp? Also does it make sense to even continue the walk if the whole loop is already moved inside the amdaie.core op?

Reply:
actually looking for at the code, I see you want and an op whose parent is the forallOp, but from the IR it looks like the consequence is the whole loop nest gets moved, so what I was suggesting would work too?

Reply:
Discussed offline. Marking resolved.